sematic/log_reader.py

```diff
@@ -422,40 +499,69 @@ def line_stream_from_log_directory(
         key=lambda path_key: int(
             path_key.replace(directory, "").replace(".log", "".replace("/", ""))
         ),
+        reverse=reverse,
     )
-    found_cursor_file = cursor_file is None
-    found_cursor_line = cursor_line_index is None
-    for log_file in log_files:
-        if log_file == cursor_file:
-            found_cursor_file = True
-        if not found_cursor_file:
-            continue
+    index_of_log_file = 0
+    if cursor_file is not None:
+        if cursor_file not in log_files:
+            raise ValueError(f"Cursor referenced invalid log file {cursor_file}.")
+        index_of_log_file = log_files.index(cursor_file)
+
+    cursor_file = log_files[index_of_log_file]
+    log_files_to_traverse = log_files[index_of_log_file:]
+    for log_file in log_files_to_traverse:
+        start_line_index = sys.maxsize if reverse else 0
```

Comment:
Given that these values appear not only in one place in the codebase, please consider extracting them to constants.

Reply:
done